### ISSUE TITLE
Can't pull records by Expedition

### ISSUE THREAD

**Issue by mbprondzinski** [ArctosDB/data-migration#156]
I was under the impression that I would be able to pull all the records from a particular Expedition... such as Expedition 41 from last summer.  Apparently this is not the case and is causing me great angst.  I need to pull records by expedition numbers, i.e. Expedition 33, Expedition 11, etc., so I can see the names of the collectors/agents and figure out how they've been entered to duplicate in the Paleo Invert records' agents.  The Expedition number was in the "Source file" field before it was loaded into Arctos.  I am not sure where it should go in the thousands of records that require it.  I believe it is showing up in the Event Remarks from the initial bulkload, but it doesn't appear to be searchable.  The  most recent "Expedition 41" name is not in those respective records, save for the Project Name.  I really need the Expedition to be searchable and linked.

**Comment by Jegelewicz** [comment]
@mbprondzinski I will look at this and see if I can get all of these in as other IDs over the holiday.

**Comment by mbprondzinski** [comment]
Sorry to spoil your holiday!

**Comment by mbprondzinski** [comment]
When I prepared the Vert Paleo data for David Bloom,I only worked on the Taxonomy and the Localities.  I never felt the data was ready for loading into Arctos and seeing how the Source File field from our data is such a mess in the original excel spreadsheet reinforces my belief we should have waited.
Both the Expeditions and the Museum Staff Trips have been entered in so many different ways, I feel I will have to go through them in the excel spreadsheet to standardize everything before you can add them to our Arctos records.  I have no idea what or how you plan on adding these on your end or what field options are available to enter these "event ID's" as I would interpret them.  They are crucial in linking the three Paleo disciplines to the same event.  
So, would it be best for me to start making standardized Event ID's for both expeditions and Museum Staff Trips BEFORE you add them to the records?   Please advise....   Thanks!

**Comment by Jegelewicz** [comment]
> So, would it be best for me to start making standardized Event ID's for both expeditions and Museum Staff Trips BEFORE you add them to the records? Please advise.... Thanks!

That might be most efficient. How hard will it be? I can probably use the old data to get soemthing started, then you can use it to standardize with OR you can create from scratch. Let me know how you want to proceed.

**Comment by mbprondzinski** [comment]
Okay, here is what I'm doing.  I have been working all  morning on standardizing the Expedition names and the Museum Staff Trip names in the "Source File" field of the old excel spreadsheet.  I hope to finish it by this afternoon, and I can then email you the file with just the Catalog Number field and the Source File field, if that is all you need?.  Anything else bad in there I can fix/add later.  How does that sound?

**Comment by Jegelewicz** [comment]
That is a start. The next question is where would they best be applied?

Other ID Number (ALMNH:Expedition)
Specimen Event Attribute (expedition identifier)

Check out https://arctos.database.museum/guid/ALMNH:ES:1962

In this record I've placed the expedition number in both a Specimen Event Attribute (red box) and an Other ID (purple box).  I can easily link the Other IDs so clicking the expedition number would immediately take you to all specimens in that expedition. The event attribute won't do that right now - but we could get that functionality eventually and you could still find everything from an event by searching on the event attribute in the main search page. Think on it and let me know which method you like.

![image](https://user-images.githubusercontent.com/5725767/69661110-07ea4880-103f-11ea-92d2-02e155c86cfe.png)

**Comment by campmlc** [comment]
Cool - I like the idea of both! Great to be implementing Specimen Event
Attributes.

**Comment by mbprondzinski** [comment]
I agree: they both are easy to spot.  I do like seeing all the Identifiers together in one box.  Just as long as it is searchable, whichever is easiest to set up.

**Comment by mbprondzinski** [comment]
[ALMNH Expeditions & Staff Trips.xlsx](https://github.com/ArctosDB/data-migration/files/3893932/ALMNH.Expeditions.Staff.Trips.xlsx)

Okay, I have standardized all the expedition records to:  "Museum Expedition ##" 
and all the staff field trips to:  "Museum Staff Trip ####"  
Feel free to change Museum to ALMNH if you think that's better.  There are over 5,300 records total.  I can't thank you enough for doing this...it will help me immensely.

(I hope it doesn't infringe too much on your holiday, Teresa!)

**Comment by mbprondzinski** [comment]
BTW, I will add Expedition 41 after the Event identifier has been added.

**Comment by Jegelewicz** [comment]
The other IDs are the easiest and probably the most helpful at this point.

**Comment by Jegelewicz** [comment]
No worries - this shouldn't take me too long.

**Comment by Jegelewicz** [comment]
OK, these are in. There are two Other ID's:

> ALMNH: Museum Expedition
> ALMNH: Museum Staff Trip

Also, I made a few changes to some of the numbers:

> - all tildes were changed to "approximately" - the tilde does not play well with Arctos
> - there was a small group of ALMNH: Museum Staff Trip IDs that had no associated year to which I assigned the identifier "unknown year"

These two records aren't in Arctos - not sure why, so I couldn't load those Other IDs:

> ALMNH:ES:5395 from ALMNH: Museum Expedition 27
> ALMNH:ES:3067 from ALMNH: Museum Staff Trip 1992

I went ahead and loaded Expedition 41 since it was easy to get the catalog numbers from the project.

This should allow you to find stuff by either of the identifiers.  I'd like to add the identifiers to the events, but that will take more thought and time. I'll keep it in my wish list of things to do and maybe it will get done in the next year. :-)

If I had all the free time in the world, I would work on standardizing your localities. The data will never be perfect, but we can plug away at it and keep a list of things we know need attention.

I ran across this one while working on this project:

![image](https://user-images.githubusercontent.com/5725767/69824207-4f501080-11c8-11ea-92a4-78682f91233c.png)

All [nine specimen records that share this Locality ID](http://arctos.database.museum/SpecimenResults.cfm?locality_id=10996767) had the same specific locality, so I assigned the locality ID to the Locality Nickname (F-26).

To see stuff from one where this isn't the case, check out [these records](http://arctos.database.museum/SpecimenResults.cfm?oidtype=Locality%20ID&oidnum=AGr-43). 

They all share a Locality ID, but there are a lot of different Specific localities.

Progress - one record at a time.....

**Comment by mbprondzinski** [comment]
Yes, this is great!  Thank you for doing this for me, Teresa.  I just haven't had the time myself to devote to the VertPaleo records since they were uploaded into Arctos.  Once I get the Invert Paleo in, I can devote more time into cleaning up and standardizing the Vert records.  I am trying to avoid as much after-clean-up with the Inverts as possible!
________________________________

**Comment by mbprondzinski** [comment]
Hi Teresa,

Happy Monday!  See my responses below…

**Comment by Jegelewicz** [comment]
> ALMNH:ES:3067 from ALMNH: Museum Staff Trip 1992  This specimen was given to the education department in 2007.

You might consider cataloging it anyway with the disposition "transfer of custody" and an associated loan with type "transfer of custody" - "Items are transferred to another institution. All or some rights may be retained as specified in the loan". This will keep a complete record of the object right in Arctos for you.

> All nine specimen records that share this Locality ID<http://arctos.database.museum/SpecimenResults.cfm?locality_id=10996767> had the same specific locality, so I assigned the locality ID to the Locality Nickname (F-26). There are some 50 “Fossil Sites” assigned by Cailup Curren for this Accession.  I’m surprised that is the only site that was recorded in the data records!  We will have to go back and add those.

I just happened upon F-26 - the other locality IDs are probably there and could be systematically checked to see if the specific localities for each site ID match or if they differ as in the example that followed this one.

> Bing’s argument is, “If it’s near that specific locality, then it gets that site number (!)

This might be OK for a "Locality ID" but a real site ID should be a single specific locality.  Something we could tease out over time and apply appropriate nicknames to the localities. We will get to it eventually!
